app/src/org/commcare/dalvik/services/CommCareSessionService.java

```diff
@@ -351,13 +354,15 @@ public void registerFormSaveCallback(FormSaveCallback callbackObj) {
 
 
     /**
-     * Conclude closing down the session by closing down key pool and user
-     * database. Performs CommCareApplication logout to unbind its connection
-     * to this object. Launches CommCareHomeActivity upon completion.
+     * Closes the key pool and user database. Performs CommCareApplication
+     * logout to unbind its connection to this object.
+     *
+     * @param sessionExpired should the user be redirected to the login screen
+     *                       upon closing this session?
      */
-    public void finishLogout() {
+    public void closeSession(boolean sessionExpired) {

```

Comment:
Fixing double login: Made closeSession take an argument to determine whether to dispatch to login screen or not